src/track/track_decl.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+/// Forward declarations for Track objects and their pointers
+
+#include <QList>
+#include <QMetaType>
+#include <memory>
+
+class Track;
+
+typedef std::shared_ptr<Track> TrackPointer;
+typedef std::weak_ptr<Track> TrackWeakPointer;
+typedef QList<TrackPointer> TrackPointerList;
+
+enum class ExportTrackMetadataResult {
```

Comment:
Remark: I have noticed that this enum class does not really belong here. Refactoring the API such that the enumeration does not belong to the `Track` object is a future task. The extraction of all definitions into the new `track_decl.h` file just revealed this issue.